libs/jsonschema/instance.go

```diff
@@ -0,0 +1,90 @@
+package jsonschema
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+// Load a JSON document, also validating it against the JSON schema. Instance here
+// refers to a JSON document. see: https://json-schema.org/draft/2020-12/json-schema-core.html#name-instance
+func (s *Schema) LoadInstance(path string) (map[string]any, error) {
+	instance := make(map[string]any, 0)
+	b, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(b, &instance)
+	if err != nil {
+		return nil, err
+	}
+
+	// The default JSON unmarshaler parses untyped number values as float64.
+	// We convert integer properties from float64 to int64 here.
```

Comment:
Ah, good point. We already have methods that convert strings to float / integers. 

Note, I actually briefly looked into this, and the source code either returns a number or a float64 explictly. Since it's a part of the stdlib, this is probably pretty stable.
source:
https://cs.opensource.google/go/go/+/master:src/encoding/json/decode.go;drc=ddad9b618cce0ed91d66f0470ddb3e12cfd7eeac;l=978
https://cs.opensource.google/go/go/+/master:src/encoding/json/decode.go;l=826;drc=729f214e3afd61afd924b946745798a8d144aad6

Created a ticket to track it internally.